### ISSUE TITLE
HiFiGAN v1 divergence on 16k Mandarin dataset

### ISSUE THREAD

**Issue by ZhaoZeqing** [TensorSpeech/TensorFlowTTS#429]
Thanks for your quickly supported on HiFiGAN!

I trained HiFiGAN v1 & v2 on our 16k Mandarin dataset. v2 convergenced but the quality is not good, v1 divergenced. My conf and loss is here:

**v1:**
sampling_rate: 16000
hop_size: 200
upsample_scales: [5, 5, 4, 2]
batch_max_steps: 12800
batch_max_steps_valid: 64000

![image](https://user-images.githubusercontent.com/8464475/102035882-86598a00-3dfc-11eb-8ea3-2048ba4b7aaf.png)

![image](https://user-images.githubusercontent.com/8464475/102035859-75a91400-3dfc-11eb-8317-4af496932a7c.png)

I tried to reduce learning rate on v1, but not worked.

**v2:**
sampling_rate: 16000
hop_size: 200
upsample_scales: [5, 5, 4, 2]

![image](https://user-images.githubusercontent.com/8464475/102035974-ba34af80-3dfc-11eb-9a08-123562df0539.png)

![image](https://user-images.githubusercontent.com/8464475/102036023-cf114300-3dfc-11eb-9622-1e47f301cc89.png)

**Comment by dathudeptrai** [comment]
> Thanks for your quickly supported on HiFiGAN!
> 
> I trained HiFiGAN v1 & v2 on our 16k Mandarin dataset. v2 convergenced but the quality is not good, v1 divergenced. My conf and loss is here:
> 
> **v1:**
> sampling_rate: 16000
> hop_size: 200
> upsample_scales: [5, 5, 4, 2]
> batch_max_steps: 12800
> batch_max_steps_valid: 64000
> 
> ![image](https://user-images.githubusercontent.com/8464475/102035882-86598a00-3dfc-11eb-8ea3-2048ba4b7aaf.png)
> 
> ![image](https://user-images.githubusercontent.com/8464475/102035859-75a91400-3dfc-11eb-8317-4af496932a7c.png)
> 
> I tried to reduce learning rate on v1, but not worked.
> 
> **v2:**
> sampling_rate: 16000
> hop_size: 200
> upsample_scales: [5, 5, 4, 2]
> 
> ![image](https://user-images.githubusercontent.com/8464475/102035974-ba34af80-3dfc-11eb-9a08-123562df0539.png)
> 
> ![image](https://user-images.githubusercontent.com/8464475/102036023-cf114300-3dfc-11eb-9622-1e47f301cc89.png)

how mb_melgan compared with hifigan-v2 ?

**Comment by ZhaoZeqing** [comment]
@dathudeptrai mb_melgan is very good, but hifigan-v2 is not so good

**Comment by dathudeptrai** [comment]
> @dathudeptrai mb_melgan is very good, but hifigan-v2 is not so good

maybe you want to try mb_hifigan_v2 :))))

**Comment by ZhaoZeqing** [comment]
@dathudeptrai Thanks! I'll try it~

**Comment by zhangsanfeng86** [comment]
@ZhaoZeqing 
Could you share some wavs  generated with hifigan?

**Comment by ZhaoZeqing** [comment]
> @ZhaoZeqing
> Could you share some wavs generated with hifigan?

I'll train hifigan with a public dataset, then share some wavs here :)

**Comment by PeijiYang** [comment]
this 16k Mandarin dataset is publication? Can you share it with me, thx.

**Comment by Jackiexiao** [comment]
> > @dathudeptrai mb_melgan is very good, but hifigan-v2 is not so good
> 
> maybe you want to try mb_hifigan_v2 :))))

should I implement it by myself or is there any code in this repo?

**Comment by dathudeptrai** [comment]
> > > @dathudeptrai mb_melgan is very good, but hifigan-v2 is not so good
> > 
> > 
> > maybe you want to try mb_hifigan_v2 :))))
> 
> should I implement it by myself or is there any code in this repo?

you can easily implement by urself :D

**Comment by Jackiexiao** [comment]
thx

**Comment by ZhaoZeqing** [comment]
> this 16k Mandarin dataset is publication? Can you share it with me, thx.

Here it is, but it may not be downloaded directly now.
https://www.data-baker.com/open_source.html

**Comment by ZDisket** [comment]
@dathudeptrai 
> maybe you want to try mb_hifigan_v2 :))))

You tried it?

**Comment by dathudeptrai** [comment]
> @dathudeptrai
> 
> > maybe you want to try mb_hifigan_v2 :))))
> 
> You tried it?

i don't remember :D

**Comment by jarred1989** [comment]
@ZhaoZeqing I ran into the same problem for Hifigan v1, have you solved it?

**Comment by ymzlygw** [comment]
> @dathudeptrai mb_melgan is very good, but hifigan-v2 is not so good

@ZhaoZeqing Hi, I am also try to train mb-melgan with 16kHZ dataset , but I don't know how to change config file in mb-melgan, can you tell me what your config is or share it if possible? Thanks a lot!

**Comment by ZhaoZeqing** [comment]
> @ZhaoZeqing I ran into the same problem for Hifigan v1, have you solved it?

It maybe the wrong lr value in config file, use the latest config file is ok:)

**Comment by ZhaoZeqing** [comment]
> > @dathudeptrai mb_melgan is very good, but hifigan-v2 is not so good
> 
> @ZhaoZeqing Hi, I am also try to train mb-melgan with 16kHZ dataset , but I don't know how to change config file in mb-melgan, can you tell me what your config is or share it if possible? Thanks a lot!

@ymzlygw I modified the **upsample_scales** in config file, the product of **upsample_scales** must be divisible by **hopsize**.

**Comment by ymzlygw** [comment]
> > > @dathudeptrai mb_melgan is very good, but hifigan-v2 is not so good
> > 
> > 
> > @ZhaoZeqing Hi, I am also try to train mb-melgan with 16kHZ dataset , but I don't know how to change config file in mb-melgan, can you tell me what your config is or share it if possible? Thanks a lot!
> 
> @ymzlygw I modified the **upsample_scales** in config file, the product of **upsample_scales** must be divisible by **hopsize**.

Thanks! I have changed it!
